### ISSUE TITLE
futureimpl.h : error C2220: warning treated as error - no 'object' file generatedDescribe the problem

### ISSUE THREAD

**Issue by felixding** [microsoft/react-native-windows#4099]
### Environment

If you are using latest version:
1. `react-native -v`:
```
react-native-cli: 2.0.1
react-native: 0.61.5
```
2. `react-native run-windows --info`:
```
  System:
    OS: Windows 10 10.0.18363
    CPU: (4) x64 Intel(R) Pentium(R) CPU 4415Y @ 1.60GHz
    Memory: 2.44 GB / 7.91 GB
  Binaries:
    Node: 13.6.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.21.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    react: 16.9.0 => 16.9.0
    react-native: 0.61.5 => 0.61.5
    react-native-windows: 0.61.0-beta.6 => 0.61.0-beta.6
    rnpm-plugin-windows: ^0.5.1-0 => 0.5.1-0
  Installed UWP SDKs:
    10.0.17763.0
    10.0.18362.0
```

Then, specify:

- Target Platform Version(s): 10.0.18362

- Target Device(s): Desktop

- Visual Studio Version: Visual Studio 2019

- Build Configuration: Debug

### Steps to Reproduce

(Write your steps here:)

1. In my project folder, run `npm install -g react-native-cli`
2. Then `npm install --save rnpm-plugin-windows`
3. Then `react-native windows --template beta`
4. Then `react-native run-windows` and I got the following errors

### Expected Behavior

Don't know. Maybe an app with UI would run?

### Actual Behavior

<img width="900" alt="Annotation 2020-02-13 174538" src="https://user-images.githubusercontent.com/111456/74416591-b5c00300-4e88-11ea-95c0-d021b14da7e8.png">

### react-native run-windows --logging

```
c:\my\projects\freedomfm-client\node_modules\react-native-windows\mso\src\future\futureimpl.h : error C2220: warning treated as
error - no 'object' file generated (compiling source file C:\my\projects\freedomfm-client\node_modules\react-native-windows\Mso\
src\future\futureImpl.cpp) [C:\my\projects\freedomfm-client\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.Re actNative.vcxproj]
c:\my\projects\freedomfm-client\node_modules\react-native-windows\mso\src\future\futureimpl.h : warning C4819: The file contains
 a character that cannot be represented in the current code page (936). Save the file in Unicode format to prevent data loss (co
mpiling source file C:\my\projects\freedomfm-client\node_modules\react-native-windows\Mso\src\future\futureImpl.cpp) [C:\my\proj ects\freedomfm-client\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]
c:\my\projects\freedomfm-client\node_modules\react-native-windows\mso\src\future\futureimpl.h : error C2220: warning treated as  error - no 'object' file generated (compiling source file C:\my\projects\freedomfm-client\node_modules\react-native-windows\Mso\ src\future\cancellationTokenImpl.cpp) [C:\my\projects\freedomfm-client\node_modules\react-native-windows\Microsoft.ReactNative\M icrosoft.ReactNative.vcxproj]
c:\my\projects\freedomfm-client\node_modules\react-native-windows\mso\src\future\futureimpl.h : warning C4819: The file contains  a character that cannot be represented in the current code page (936). Save the file in Unicode format to prevent data loss (co mpiling source file C:\my\projects\freedomfm-client\node_modules\react-native-windows\Mso\src\future\cancellationTokenImpl.cpp)  [C:\my\projects\freedomfm-client\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]
```

### Reproducible Demo

Well, actually the issue occurs even when I follow the docs and try to create a demo app.

**Comment by NickGerleman** [comment]
Oh huh, looks like our build isn't resilient to a different locale. 

We could add a BOM to the file or remove the character, but it might be better to just [tell MSVC to use UTF-8](https://docs.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8?view=vs-2019). We don't want to silently break if someone adds an emoji to a unit test.

**Comment by NickGerleman** [comment]
I'll try to reproduce and fix this today with the mentioned system character set.

**Comment by NickGerleman** [comment]
I believe this should be fixed in the newest beta version we're publishing. I wasn't able to reproduce the issue though, so I cannot 100% confirm the fix.

**Comment by NickGerleman** [comment]
The fix is in `react-native-windows@0.61.0-beta.7`. Please feel free to reopen the issue if we still see it in that build or later.
